src/utils/FixedPointMath.sol

```diff
@@ -94,16 +94,28 @@ library FixedPointMath {
                           GENERAL NUMBER UTILS
     //////////////////////////////////////////////////////////////*/
 
-    function sqrt(uint256 y) internal pure returns (uint256 z) {
-        if (y > 3) {
-            z = y;
-            uint256 x = y / 2 + 1;
-            while (x < z) {
-                z = x;
-                x = (y / x + x) / 2;
-            }
-        } else if (y != 0) {
-            z = 1;
+    // Credit to ABDK Consulting (https://github.com/abdk-consulting/abdk-libraries-solidity/blob/master/ABDKMath64x64.sol)
+    function sqrt(uint256 x) internal pure returns (uint256 z) {
```

Comment:
isn't all this logic wrapped in an unchecked block in ABDK's impl?

Reply:
Not in the ABDK impl but it is in the HiFi-Finance impl: https://github.com/hifi-finance/prb-math/blob/main/contracts/PRBMath.sol

Reply:
nope jk it is in the ABDK impl: https://github.com/abdk-consulting/abdk-libraries-solidity/blob/master/ABDKMath64x64.sol

will update